src/main/java/com/dev/cinema/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.exeptions.DataProcessingException;
+import com.dev.cinema.library.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
see checklist: Use try with finally, where you use transaction

Reply:
done